centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1470,30 +1490,29 @@ objDataToMoney(int data_type, void* data_ptr, pMoneyType m)
 		    }
 		if (*endptr == '-')
 		    {
-		    m->WholePart = -m->WholePart;
+		    m->MoneyValue = -m->MoneyValue;
 		    is_neg = !is_neg;
 		    }
-		if (is_neg && m->FractionPart != 0)
-		    {
+		/*if (is_neg && m->FractionPart != 0)
+		    { //This handled the unsigned representation of FractionPart
 		    m->WholePart--;
 		    m->FractionPart = 10000 - m->FractionPart;
-		    }
+		    }*/
 		break;
 	    
 	    case DATA_T_DOUBLE:
 	        dbl = *(double*)data_ptr + 0.000001;
-	        m->WholePart = floor(dbl);
-		m->FractionPart = (dbl - m->WholePart)*10000;
+	        long long WholePart = floor(dbl);
+		    int FractionPart = (dbl - WholePart)*10000;
+		    m->MoneyValue = (WholePart*10000) + FractionPart;
 		break;
 	
 	    case DATA_T_INTEGER:
-	        m->WholePart = *(int*)data_ptr;
-		m->FractionPart = 0;
+	        m->MoneyValue = *(long long*)data_ptr;
```

Comment:
You should multiply the int by 10000 so it represents the whole part, as in the original version